frame/system/src/lib.rs

```diff
@@ -164,378 +92,136 @@ pub fn extrinsics_data_root<H: Hash>(xts: Vec<Vec<u8>>) -> H::Output {
-#[cfg(feature = "std")]
-fn hash69<T: AsMut<[u8]> + Default>() -> T {
-	let mut h = T::default();
-	h.as_mut().iter_mut().for_each(|byte| *byte = 69);
-	h
-}
-
-/// This type alias represents an index of an event.
-///
-/// We use `u32` here because this index is used as index for `Events<T>`
-/// which can't contain more than `u32::max_value()` items.
-type EventIndex = u32;
-
-/// Type used to encode the number of references an account has.
-pub type RefCount = u32;
-
-/// Information of an account.
-#[derive(Clone, Eq, PartialEq, Default, RuntimeDebug, Encode, Decode)]
-pub struct AccountInfo<Index, AccountData> {
-	/// The number of transactions this account has sent.
-	pub nonce: Index,
-	/// The number of other modules that currently depend on this account's existence. The account
-	/// cannot be reaped until this is zero.
-	pub refcount: RefCount,
-	/// The additional data that belongs to this account. Used to store the balance(s) in a lot of
-	/// chains.
-	pub data: AccountData,
-}
-
-/// Stores the `spec_version` and `spec_name` of when the last runtime upgrade
-/// happened.
-#[derive(sp_runtime::RuntimeDebug, Encode, Decode)]
-#[cfg_attr(feature = "std", derive(PartialEq))]
-pub struct LastRuntimeUpgradeInfo {
-	pub spec_version: codec::Compact<u32>,
-	pub spec_name: sp_runtime::RuntimeString,
-}
-
-impl LastRuntimeUpgradeInfo {
-	/// Returns if the runtime was upgraded in comparison of `self` and `current`.
-	///
-	/// Checks if either the `spec_version` increased or the `spec_name` changed.
-	pub fn was_upgraded(&self, current: &sp_version::RuntimeVersion) -> bool {
-		current.spec_version > self.spec_version.0 || current.spec_name != self.spec_name
-	}
-}
-
-impl From<sp_version::RuntimeVersion> for LastRuntimeUpgradeInfo {
-	fn from(version: sp_version::RuntimeVersion) -> Self {
-		Self {
-			spec_version: version.spec_version.into(),
-			spec_name: version.spec_name,
-		}
-	}
-}
-
-decl_storage! {
-	trait Store for Module<T: Config> as System {
-		/// The full account information for a particular account ID.
-		pub Account get(fn account):
-			map hasher(blake2_128_concat) T::AccountId => AccountInfo<T::Index, T::AccountData>;
-
-		/// Total extrinsics count for the current block.
-		ExtrinsicCount: Option<u32>;
-
-		/// The current weight for the block.
-		BlockWeight get(fn block_weight): ConsumedWeight;
-
-		/// Total length (in bytes) for all extrinsics put together, for the current block.
-		AllExtrinsicsLen: Option<u32>;
-
-		/// Map of block numbers to block hashes.
-		pub BlockHash get(fn block_hash) build(|_| vec![(T::BlockNumber::zero(), hash69())]):
-			map hasher(twox_64_concat) T::BlockNumber => T::Hash;
-
-		/// Extrinsics data for the current block (maps an extrinsic's index to its data).
-		ExtrinsicData get(fn extrinsic_data): map hasher(twox_64_concat) u32 => Vec<u8>;
-
-		/// The current block number being processed. Set by `execute_block`.
-		Number get(fn block_number): T::BlockNumber;
+pub use pallet::*;
```

Comment:
Will this duplicate all the documentation I wonder?

Reply:
pallet module is public, thus it will just show the re-export as in https://crates.parity.io/sc_service/index.html#reexports

sidenote: the reason for it is that in node/runtime/src/lib.rs people can use `frame_system` in construct_runtime.
Otherwise they have to do:
```
use frame_system::pallet as pallet_frame_system;

construct_runtime!(
...
System: pallet_frame_system::....
)
```
So to avoid breaking change I think it is best to reexport the module here.

Reply:
I confirm that it just shows up in the reexports section